faust-build/src/lib.rs

```diff
@@ -44,178 +60,168 @@ impl Default for FaustBuilder {
 }
 
 impl FaustBuilder {
-    pub fn new(in_file: &str, out_file: &str) -> Self {
+    pub fn new(in_file: impl Into<PathBuf>, out_file: impl Into<PathBuf>) -> Self {
         Self {
-            in_file: in_file.to_string(),
-            out_file: out_file.to_string(),
+            in_file: in_file.into(),
+            out_file: out_file.into(),
             ..Default::default()
         }
     }
 
-    pub fn set_struct_name(mut self, struct_name: &str) -> Self {
-        self.struct_name = Some(struct_name.to_string());
+    #[must_use]
+    pub fn set_struct_name(mut self, struct_name: impl Into<String>) -> Self {
+        self.struct_name = Some(struct_name.into());
         self
     }
-    pub fn set_module_name(mut self, module_name: &str) -> Self {
-        self.module_name = module_name.to_string();
+    #[must_use]
+    pub fn set_module_name(mut self, module_name: impl Into<String>) -> Self {
+        self.module_name = module_name.into();
         self
     }
 
+    #[must_use]
     pub fn set_use_double(mut self, use_double: bool) -> Self {
         self.use_double = use_double;
         self
     }
 
-    pub fn set_faust_path(mut self, faust_path: &str) -> Self {
-        self.faust_path = Some(faust_path.to_string());
+    #[must_use]
+    pub fn set_faust_path(mut self, faust_path: impl Into<PathBuf>) -> Self {
+        self.faust_path = Some(faust_path.into());
         self
     }
-    pub fn set_arch_file(mut self, arch_file: &str) -> Self {
-        self.arch_file = Some(arch_file.to_string());
+    #[must_use]
+    pub fn set_arch_file(mut self, arch_file: impl Into<PathBuf>) -> Self {
+        self.arch_file = Some(arch_file.into());
         self
     }
+
     /// Add additionals args to the faust build command
-    pub fn faust_arg(mut self, arg: &str) -> Self {
-        self.faust_args.push(arg.to_string());
+    #[must_use]
+    pub fn faust_arg(mut self, arg: impl Into<String>) -> Self {
```

Comment:
Wouldn't it be nicer to have Into<FaustArg> here? To keep it simply We could map every &str/String to FaustArg::Custom for now